### PR TITLE
Update to use Guzzle 6

### PR DESCRIPTION
**Pull request by chadicus**

### CODE REVIEW

**Comment by chrisryan** [comment]
:+1: I tested this on two applications and did not see any problems using this.

**Comment by chrisryan** [comment]
Required changes on both so this is definitely backward breaking.

**Comment by chrisryan** [comment]
Did you want to update the coveralls to the latest version as well?

**Comment by chadicus** [comment]
@chrisryan I made made some significant changes. Made use of [async requests](http://docs.guzzlephp.org/en/latest/quickstart.html#async-requests) so we can keep the multi request functionality.

File: src/GuzzleAdapter.php [hunk 1]
```diff
@@ -33,26 +42,33 @@
...
-            $request->getHeaders(),
-            $request->getBody(),
-            //stop guzzle from following redirects
-            //only for 400/500 error codes, actual exceptions can still happen
-            ['allow_redirects' => false, 'exceptions' => false]
+            [
+                'headers' => $request->getHeaders(),
+                'body' => $request->getBody(),
+            ],
+            [
```
Comment (chrisryan) [inline src/GuzzleAdapter.php]: requestAsync does not take 4 arguments according to the docs.

**Comment by chrisryan** [comment]
:+1: Both my projects tested without error against this code.

**Pull request merged**